python/tank/descriptor/descriptor.py

```diff
@@ -141,6 +66,123 @@ class Descriptor(object):
 
     (APP, FRAMEWORK, ENGINE, CONFIG, CORE, INSTALLED_CONFIG) = range(6)
 
+    @classmethod
+    def create(
```

Comment:
Why what this changed, you might ask?

This has been bugging me for a while. In order to create a descriptor, one has to


from sgtk.descriptor import create_descriptor, Descriptor, because you need one method to create the objects, but also the Descriptor class to access the descriptor type. With this backwards compatible change to the API, we can now do

```
from sgtk.descriptor import Descriptor
Descriptor.create(sg, Descriptor.CORE,...)
```

This feels much nicer.